src/node/internal/streams_duplex.js

```diff
@@ -67,32 +70,80 @@ Object.setPrototypeOf(Duplex, Readable);
   }
 }
 
-export function isDuplexInstance(obj) {
-  return obj instanceof Duplex;
-}
+// Use the `destroy` method of `Writable`.
+Duplex.prototype.destroy = Writable.prototype.destroy;
 
 export function Duplex(options) {
   if (!(this instanceof Duplex)) return new Duplex(options);
-  Readable.call(this, options);
-  Writable.call(this, options);
+
+  this._events ??= {
+    close: undefined,
+    error: undefined,
+    prefinish: undefined,
+    finish: undefined,
+    drain: undefined,
+    data: undefined,
+    end: undefined,
+    readable: undefined,
+    // Skip uncommon events...
+    // pause: undefined,
+    // resume: undefined,
+    // pipe: undefined,
+    // unpipe: undefined,
+    // [destroyImpl.kConstruct]: undefined,
+    // [destroyImpl.kDestroy]: undefined,
+  };
+
+  this._readableState = new Readable.ReadableState(options, this, true);
+  this._writableState = new Writable.WritableState(options, this, true);
+
   if (options) {
     this.allowHalfOpen = options.allowHalfOpen !== false;
+
     if (options.readable === false) {
       this._readableState.readable = false;
       this._readableState.ended = true;
       this._readableState.endEmitted = true;
     }
+
     if (options.writable === false) {
       this._writableState.writable = false;
       this._writableState.ending = true;
       this._writableState.ended = true;
       this._writableState.finished = true;
     }
+
+    if (typeof options.read === 'function') this._read = options.read;
+
+    if (typeof options.write === 'function') this._write = options.write;
+
+    if (typeof options.writev === 'function') this._writev = options.writev;
+
+    if (typeof options.destroy === 'function') this._destroy = options.destroy;
+
+    if (typeof options.final === 'function') this._final = options.final;
+
+    if (typeof options.construct === 'function')
+      this._construct = options.construct;
+
+    if (options.signal) {
+      addAbortSignal(options.signal, this);
+    }
   } else {
     this.allowHalfOpen = true;
   }
+
+  Stream.call(this, options);
+
+  if (this._construct != null) {
+    destroyConstruct(this, () => {
```

Comment:
Constructor is the only and the most important place to check @jasnell 

Particularly this line fixes  the race condition of throwing AbortErrors and catching it with await once...